### PR TITLE
hibernate hw

### PR DESCRIPTION
**Pull request by Taniann**

### CODE REVIEW

File: src/main/java/ua/mateacademy/tania/lesson24/hibernate/OrderDaoImpl.java [hunk 1]
```diff
@@ -79,7 +79,7 @@ public Order findOrderById(BigDecimal id) {
             // Getting Transaction Object From Session Object
             sessionObj.beginTransaction();
 
-            order = sessionObj.load(Order.class, id);
+            order = sessionObj.get(Order.class, id);
```
Comment (mkharkhuta) [inline src/main/java/ua/mateacademy/tania/lesson24/hibernate/OrderDaoImpl.java:82]: there is no closed session

File: src/main/java/ua/mateacademy/tania/lesson24/hibernate/OrderDaoImpl.java [hunk 2]
```diff
@@ -92,17 +92,87 @@ public Order findOrderById(BigDecimal id) {
...
+                LOG.info("\n.......Transaction Is Being Rolled Back.......\n");
+                sessionObj.getTransaction().rollback();
+            }
+            sqlException.printStackTrace();
+        } finally {
+            if (sessionObj != null) {
+                sessionObj.close();
+            }
+        }
+        return result;
```
Comment (mkharkhuta) [inline src/main/java/ua/mateacademy/tania/lesson24/hibernate/OrderDaoImpl.java:175]: where is the getAll method?)

Comment (mkharkhuta) [reply src/main/java/ua/mateacademy/tania/lesson24/hibernate/OrderDaoImpl.java:175]: found already)

**Review by mkharkhuta — APPROVED** [review]
close one session, and everything else is good)

File: src/test/java/ua/mateacademy/tania/lesson24/hibernate/OrderDaoImplTest.java [hunk 3]
```diff
@@ -1,39 +1,62 @@
...
+    private static final Order ORDER = new Order(BigDecimal.valueOf(11111111), CUSTOMER, null, null, new Date(),
+                                            "mfr", BigDecimal.valueOf(1000), BigDecimal.valueOf(1000));
     private OrderDao orderDao = new OrderDaoImpl();
 
     @Test
-    public void getAllOrder() {
-        System.out.println(orderDao.getAllOrder());
+    public void testGetAllOrder() {
+        Set<Order> orders = orderDao.getAllOrder();
+        System.out.println(orders);
```
Comment (SerhiiPasko) [inline src/test/java/ua/mateacademy/tania/lesson24/hibernate/OrderDaoImplTest.java:30]: In real tests we don't write sout

File: src/test/java/ua/mateacademy/tania/lesson24/hibernate/OrderDaoImplTest.java [hunk 4]
```diff
@@ -1,39 +1,62 @@
...
-        System.out.println(order);
+    public void testFindOrderById() {
+        Order order = orderDao.findOrderById(ALREADY_EXIST_ORDER);
         assertNotNull(order);
     }
 
     @Test
-    public void insertOrder() {
+    public void testFindOrderByIdReturnsNull() {
+        Order order = orderDao.findOrderById(NOT_EXIST_ORDER);
```
Comment (SerhiiPasko) [inline src/test/java/ua/mateacademy/tania/lesson24/hibernate/OrderDaoImplTest.java:43]: There you will have very interesting situation when close session

**Review by SerhiiPasko — CHANGES_REQUESTED** [review]
Read comments

File: src/test/java/ua/mateacademy/tania/lesson24/hibernate/OrderDaoImplTest.java (hunk 4 above)
Comment (Taniann) [reply src/test/java/ua/mateacademy/tania/lesson24/hibernate/OrderDaoImplTest.java:43]: > There you will have very interesting situation when close session

I`ve corrected method findOrderById. now session is closing in finally block. but I didnt notice interesting situation here

**Pull request merged**